components/brave_wallet_ui/common/slices/endpoints/nfts.endpoints.ts

```diff
@@ -0,0 +1,249 @@
+      boolean, // success
+      boolean
+    >({
+      queryFn: async (arg, _api, _extraOptions, _baseQuery) => {
+        try {
+          const { braveWalletService } = apiProxyFetcher()
+          await braveWalletService.setNftDiscoveryEnabled(arg)
+
+          return {
+            data: true
+          }
+        } catch (error) {
+          return {
+            error: 'Failed to set NFT auto-discovery status'
+          }
+        }
+      },
+      invalidatesTags: ['NftDiscoveryEnabledStatus']
+    }),
+    getNftMetadata: query<NFTMetadataReturnType, BraveWallet.BlockchainToken>({
+      queryFn: async (arg, _api, _extraOptions, baseQuery) => {
+        try {
+          const { data: api } = baseQuery(undefined)
+          const { jsonRpcService } = api
+          const result =
+            arg.coin === BraveWallet.CoinType.ETH
+              ? await jsonRpcService.getERC721Metadata(
+                  arg.contractAddress,
+                  arg.tokenId,
+                  arg.chainId
+                )
+              : arg.coin === BraveWallet.CoinType.SOL
+              ? await jsonRpcService.getSolTokenMetadata(
+                  arg.chainId,
+                  arg.contractAddress
+                )
+              : undefined
+
+          if (!result?.error) {
+            const response = result?.response && JSON.parse(result.response)
+            const attributes = Array.isArray(response.attributes)
+              ? response.attributes.map(
+                  (attr: { trait_type: string; value: string }) => ({
+                    traitType: attr.trait_type,
+                    value: attr.value
+                  })
+                )
+              : []
+            const tokenNetwork = await getNetwork(api, arg)
+            const nftMetadata: NFTMetadataReturnType = {
+              metadataUrl: result?.tokenUrl || '',
+              chainName: tokenNetwork?.chainName || '',
+              tokenType:
+                arg.coin === BraveWallet.CoinType.ETH
+                  ? 'ERC721'
+                  : arg.coin === BraveWallet.CoinType.SOL
+                  ? 'SPL'
+                  : '',
+              tokenID: arg.tokenId,
+              imageURL: response.image.startsWith('data:image/')
+                ? response.image
+                : await translateToNftGateway(response.image),
+              imageMimeType: 'image/*',
+              floorFiatPrice: '',
+              floorCryptoPrice: '',
+              contractInformation: {
+                address: arg.contractAddress,
+                name: response.name,
+                description: response.description,
+                website: '',
+                facebook: '',
+                logo: '',
+                twitter: ''
+              },
+              attributes
+            }
+
+            return {
+              data: nftMetadata
+            }
+          } else {
+            throw new Error(result.errorMessage)
+          }
```

Comment:
nit: this might look cleaner if you throw early. that way the bulk of the logic wont be wrapped in the `if`, and you wont need an `else`